internal/api_server/server.go

```diff
@@ -68,6 +68,13 @@ func oapiErrorHandler(w http.ResponseWriter, message string, statusCode int) {
 	http.Error(w, fmt.Sprintf("API Error: %s", message), statusCode)
 }
 
+const (
+	readTimeout    = time.Second * 5 // Set request read timeout to 5 seconds
```

Comment:
Question: what happens when the server is under resource starvation and this timeout is met? What is the advantage of having this set?

Reply:
This is a good article[1] on the topic they recommend setting `ReadHeaderTimeout` vs `ReadTimeout`. I just want to make sure this is what we want.

[1] https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6

Reply:
Brad notes `ReadTimeout` but this is pretty old[1].

[1] https://groups.google.com/g/golang-nuts/c/MFZd6b8zQTQ/m/ISD96eyXtD4J